### PR TITLE
Bump up version in main from 1.18.0 to 1.19.0

### PR DESCRIPTION
**Pull request by yihonglyu**
Bump up version in main from 1.18.0 to 1.19.0 since the release branch has been cut.

### CODE REVIEW

**Review by fs-eire — APPROVED** [review]
approved for js.

File: onnxruntime/core/session/onnxruntime_c_api.cc [hunk 1]
```diff
@@ -2758,16 +2758,16 @@ static_assert(offsetof(OrtApi, KernelContext_GetResource) / sizeof(void*) == 265
...
 
 // So that nobody forgets to finish an API version, this check will serve as a reminder:
-static_assert(std::string_view(ORT_VERSION) == "1.18.0",
+static_assert(std::string_view(ORT_VERSION) == "1.19.0",
               "ORT_Version change detected, please follow below steps to ensure OrtApi is updated properly");
 // 1. Update the hardcoded version string in above static_assert to silence it
-// 2. If there were any APIs added to ort_api_1_to_18 above:
+// 2. If there were any APIs added to ort_api_1_to_19 above:
 //    a. Add the 'End of version #' markers (pattern above should be obvious)
 //    b. Add a static_assert in the directly above list of version sizes to ensure nobody adds any more functions to the just shipped API version
```
Comment (edgchen1) [inline onnxruntime/core/session/onnxruntime_c_api.cc:2766]: do these steps need to be done?

**Review by sophies927 — DISMISSED** [review]
LGTM

File: onnxruntime/core/session/onnxruntime_c_api.cc (hunk 1 above)
Comment (YUNQIUGUO) [reply onnxruntime/core/session/onnxruntime_c_api.cc:2766]: good catch - yes, please update/add static_assert for the newly added/just shipped APIs in this version.

File: onnxruntime/core/session/onnxruntime_c_api.cc [hunk 2]
```diff
@@ -2723,10 +2723,14 @@ static constexpr OrtApi ort_api_1_to_18 = {
     &OrtApis::SetDeterministicCompute,
     &OrtApis::KernelContext_ParallelFor,
     &OrtApis::SessionOptionsAppendExecutionProvider_OpenVINO_V2,
+    // End of Version 17 - DO NOT MODIFY ABOVE (see above text for more information)
+
     &OrtApis::SessionOptionsAppendExecutionProvider_VitisAI,
     &OrtApis::KernelContext_GetScratchBuffer,
     &OrtApis::KernelInfoGetAllocator,
-    &OrtApis::AddExternalInitializersFromFilesInMemory};
+    &OrtApis::AddExternalInitializersFromFilesInMemory
```
Comment (edgchen1) [inline onnxruntime/core/session/onnxruntime_c_api.cc:2731]: nit: so future additions don't need to update this line

```suggestion
    &OrtApis::AddExternalInitializersFromFilesInMemory,
```

**Review by edgchen1 — APPROVED** [review]

**Review by sophies927 — APPROVED** [review]

**Review by YUNQIUGUO — APPROVED** [review]

**Review by edgchen1 — APPROVED** [review]

**Pull request merged**